pkg/log/zap/zap.go

```diff
@@ -181,12 +181,17 @@ func (o *Options) addDefaults() {
 			lvl := zap.NewAtomicLevelAt(zap.ErrorLevel)
 			o.StacktraceLevel = &lvl
 		}
-		o.ZapOpts = append(o.ZapOpts,
-			zap.WrapCore(func(core zapcore.Core) zapcore.Core {
-				return zapcore.NewSampler(core, time.Second, 100, 100)
-			}))
+		// Disable sampling when we are in debug mode. Otherwise, this will
+		// cause index out of bounds errors in the sampling code.
+		if o.Level.Enabled(zapcore.DebugLevel) {
+			o.ZapOpts = append(o.ZapOpts, zap.Development())
+		} else {
+			o.ZapOpts = append(o.ZapOpts,
+				zap.WrapCore(func(core zapcore.Core) zapcore.Core {
+					return zapcore.NewSampler(core, time.Second, 100, 100)
+				}))
+		}
```

Comment:
Wouldn't this only cover `Level(-2)` ,We still need to handle `Level(-3)` or below.

Reply:
Agree , We can remove the developement mode, and it can be as shown below.
```Go
	// Disable sampling when we are in debug mode. Otherwise, this will
		// cause index out of bounds errors in the sampling code.
		if o.Level.Enabled(zapcore.DebugLevel) {
			o.ZapOpts = append(o.ZapOpts)
		} else {
			o.ZapOpts = append(o.ZapOpts,
				zap.WrapCore(func(core zapcore.Core) zapcore.Core {
					return zapcore.NewSampler(core, time.Second, 100, 100)
				}))
		}
```